lib/puppet/indirector/report/rest.rb

```diff
@@ -6,6 +6,16 @@ class Puppet::Transaction::Report::Rest < Puppet::Indirector::REST
   use_port_setting(:report_port)
   use_srv_service(:report)
 
+  def save(request)
+    super(request) do |response|
+      server_version = response['X-Puppet-Version']
+      if !response.is_a?(Net::HTTPSuccess) && SemanticPuppet::Version.parse(server_version).major < SemanticPuppet::Version.parse(Puppet::PUPPETVERSION).major
+        Puppet.err "Server version #{server_version} is unsupported, set `preferred_serialization_format=pson`"
```

Comment:
Maybe worth noting that this failure is in the context of sending the report since during an agent run it can difficult to tell what stage we're in